lte/gateway/c/core/oai/tasks/amf/amf_metadata_util.cpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "include/amf_metadata_util.h"
+#include "include/amf_client_servicer.h"
+#include <memory>
+
+std::shared_ptr<amf_metadata_t> metadata_info_ptr{};
+
+/* Function to initialize amf metadata information */
+void amf_metadata_initialize(
+    const std::shared_ptr<amf_metadata_t>& metadata_p) {
+  metadata_info_ptr = std::move(metadata_p);
+
+  metadata_info_ptr->amf_client_servicer_init();
+}
+
+std::shared_ptr<magma5g::AmfClientServicer> amf_get_client_servicer_ref() {
+  return metadata_info_ptr->amf_client_servicer;
+}
```

Comment:
The original comment was about to eliminate globals. The pattern should be like the following:
1. Create meta data object during task initialization.
2. Have new function `init_task_context_with_args`  (in the same file as `init_task_context`) and pass the meta data  pointer as the arguments for handler function. Inside `init_task_context_with_args`, pass the argument to zloop_reader.
3. In handler function, pass the pointer for metadata to called functions if they need to access the meta data.

cc @ardzoht

Reply:
Tried making the change. Metadata parameter will be used by function : start_authentication_information_procedure().  This function is coming at a stack depth of 16. So the changes are spanning across many files/functions.  Need to discuss on the same.